openedx_external_enrollments/tests/external_enrollments/tests_external_enrollments.py

```diff
@@ -0,0 +1,57 @@
+"""Tests External enrollments file."""
+import logging
+
+from django.test import TestCase
+from mock import Mock, patch
+from testfixtures import LogCapture
+
+from openedx_external_enrollments.external_enrollments import execute_external_enrollment
+
+module = 'openedx_external_enrollments.external_enrollments'
+
+
+class ExecuteExternalEnrollmentTest(TestCase):
+    """Test class for execute_external_enrollment method."""
+
+    @patch('openedx_external_enrollments.external_enrollments.configuration_helpers')
+    def test_execute_external_enrollment(self, configuration_helpers_mock):
+        """Testing execute_external_enrollment method."""
+        course = Mock()
+        course.id = 'test-course-id'
+        data = {'fake': 'data'}
+        course.other_course_settings.get.return_value = 'edx'
+        configuration_helpers_mock.get_value.return_value = ['openedx']
+
+        with LogCapture(level=logging.WARNING) as log_capture:
+            execute_external_enrollment(data, course)
+            log = 'The controller {} is not present in the valid external targets list {}.'.format(
+                'edx',
+                ['openedx'],
+            )
+            log_capture.check(
+                (module, 'WARNING', log),
+            )
+
+        with patch('openedx_external_enrollments.external_enrollments.ExternalEnrollmentFactory') as factory_mock:
```

Comment:
I like to keep the relation one-one, that means that I will create one test method per every method that I have, I don't like to create a test method per every possible scenario because that means create a lot of methods, for example if I have the following method I would have to create 26 test cases that do the same 

```
def m(key):
   if key == 'a':
        return 1
   if key == 'b':
        return 2
   if key == 'c':
        return 3
    .
    .
    .
   if key == 'z':
        return 26
```